server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1834,6 +1836,42 @@ public void syncDatastoreClusterStoragePool(long datastoreClusterPoolId, List<Mo
         handleRemoveChildStoragePoolFromDatastoreCluster(childDatastoreUUIDs);
     }
 
+    /**
+     * fixed mismatching between db uuids and and custom
+     * attribute uuids
+     *
+     * To different formats of uuids exists
+     * 1. xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+     * 2. xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+     * @param uuid of existing pool
+     * @return existing pull or null otherwise
+     */
+    private StoragePoolVO getExistingPoolByUuid(String uuid){
+        StoragePoolVO storagePool = _storagePoolDao.findByUuid(uuid);
+        if(storagePool != null){
+            return storagePool;
+        }
+
+        //this case is unlikely (DB uuid without separators), but safety first
+        if(uuid.contains("-")){
+            uuid = uuid.replaceAll("-", "");
+            storagePool = _storagePoolDao.findByUuid(uuid);
+            if(storagePool != null){
+                return storagePool;
+            }
+        }
+
+        //transform uuid in the valid format with separators
+        UUID poolUuid;
+        poolUuid = new UUID(
+                new BigInteger(uuid.substring(0, 16), 16).longValue(),
+                new BigInteger(uuid.substring(16), 16).longValue()
+        );
+
+        return _storagePoolDao.findByUuid(poolUuid.toString());
+    }
```

Comment:
```suggestion
    private StoragePoolVO getExistingPoolByUuid(String uuid){
        String removedSepators = uuid.contains("-") ? uuid.replaceAll("-", "") : null;
        String formatted = new UUID(
                new BigInteger(uuid.substring(0, 16), 16).longValue(),
                new BigInteger(uuid.substring(16), 16).longValue()).toString();
        List<String> options = Arrays.asList(uuid, removedSepators, formatted);
        return getFirstValidStoragePoolFromUuidOptions(options);
    }

    private StoragePoolVO getFirstValidStoragePoolFromUuidOptions(List<String> uuids) {
        for (String uuid : uuids) {
            if (StringUtils.isBlank(uuid)) {
                continue;
            }
            StoragePoolVO pool = _storagePoolDao.findByUuid(uuid);
            if (pool != null) {
                return pool;
            }
        }
        return null;
    }
```

I think something like this could help readability. Can also add unit tests for the new method passing different options as a parameter

Reply:
@nvazquez I think that's a matter of taste :D. You make two actions in getExistingPoolByUuid that are not always necessary. But I also find your code quite nice and I will write a unit test.

@DaanHoogland I thought I had done that with the method :D. 
Not quite sure what you mean.

Reply:
what i mean is rigorously remove all '-'s and toLower() both strings and compare the results but not touch the originals.

Reply:
never mind my comment. I think @sureshanaparti 's comment at https://github.com/apache/cloudstack/pull/5382#discussion_r698256812 is more important.